api/slack.go

```diff
@@ -6,24 +6,31 @@ import (
 	"github.com/slack-go/slack"
 )
 
-func SendSlackMessages(slackToken string, messages map[string]string) {
-	log := logger.Get()
+type SlackClientInterface interface {
+	PostMessage(channelID string, options ...slack.MsgOption) (string, string, error)
+}
 
-	if len(slackToken) > 0 {
-		slackClient := slack.New(slackToken, slack.OptionDebug(true))
-		for channel, message := range messages {
-			_, timestamp, err := slackClient.PostMessage(
-				channel,
-				slack.MsgOptionText(message, false),
-				slack.MsgOptionAsUser(true),
-			)
+type SlackClient struct {
+	client SlackClientInterface
+}
+
+func (s *SlackClient) PostMessage(channelID string, options ...slack.MsgOption) (string, string, error) {
+	return s.client.PostMessage(channelID, options...)
+}
 
-			if err != nil {
-				log.Error().Err(err).Msg("Failed to send Slack message.")
-			}
-			log.Info().Str("channel", channel).Str("timestamp", timestamp).Msg("Message sent to Slack.")
+func NewSlackClient(slackToken string) SlackClientInterface {
+	return &SlackClient{
+		client: slack.New(slackToken, slack.OptionDebug(true)),
+	}
+}
+
+func SendSlackMessages(messages map[string]string, client SlackClientInterface) {
+	log := logger.Get()
+	for channel, message := range messages {
+		_, timestamp, err := client.PostMessage(channel, slack.MsgOptionText(message, false), slack.MsgOptionAsUser(true))
+		if err != nil {
+			log.Error().Err(err).Msg("Failed to send Slack message.")
 		}
-	} else {
-		log.Warn().Msg("No Slack token found. Skipping communication.")
+		log.Info().Str("channel", channel).Str("timestamp", timestamp).Msg("Message sent to Slack.")
```

Comment:
It looks like this will now always be logged, even if the sending fails. Should this be inside an `else` block?

Reply:
You're right, fixing!